src/fiber_util/fiber_util.mli

```diff
@@ -62,3 +62,87 @@ module Cancellation : sig
     -> on_cancellation:(unit -> 'b Fiber.t)
     -> ('a * 'b outcome) Fiber.t
 end
+
+(** A observer for a value that's changing over time *)
+module Observer : sig
+  (** An observer is something that let's one monitor a value that is changing
+      over time.
+
+      We call a value that is changing over time and can be observed an
+      "observable". An "observer" is a routine watching this observable and
+      getting notified when the value change. An observable may have several
+      observers.
+
+      The goal of an observer is to be up-to-date with the observable. It is not
+      guaranteed that the observer will observe all possible intermediate values
+      of the observable. For instance, if an observable changes several time in
+      a row, by the time the observer is awoken it will receive the latest value
+      of the observable. {!await} will only suspend the current fiber if the
+      current value of the observable is the same as the last one that was
+      reported.
+
+      Given that an observer is stateful, it shouldn't be shared between
+      different consumers. Each consumer that wants to watch an observable
+      should create it's wn observer. Sharing an observer would lead to
+      consumers not being up-to-date with the latest value of the observable. An
+      observer should always be unsubscribed to via {!unsubscribe} when done
+      with, otherwise the observable will continue updating it.
+
+      Finally, sometimes the value is big but only changes a little at a time.
+      For instance, an observable might represent "the set of current errors",
+      and the producer will only add a remove one error at time. To avoid
+      dumping a massive value on each update, we might instead want to send
+      small "diffs". In such a case, values reported by {!await} would be diffs
+      rather than full values and it is up to the consumer to reconstruct the
+      full value if they want it. Successive diffs would still be squashed so
+      that a call to {!await} returns them all at once. *)
```

Comment:
Shall this last paragraph be moved to `Observable`? The API of `Observer` doesn't seem to care about diffs at the moment (which is perhaps a bad thing: what if the observer wants to process diffs too?).

Reply:
Oh, perhaps, I misunderstand the current API: can `'a` stand both for a value as well as for a diff?

Reply:
If this is just a pipe with batching (`combine`) then I think we should just call it that to avoid others from getting confused too.

Reply:
> Oh, perhaps, I misunderstand the current API: can 'a stand both for a value as well as for a diff?

Yes. I wasn't 100% sure about this either. If you got confused as well, then that's an indication the API is confusing and we should have two type parameters:

- one for values
- one for diffs

> If that's the case then I don't see how this is different from a pipe with diffs.

It has one write end and multiple read end. It's some kind of "forked pipe".

> If this is just a pipe with batching (combine) then I think we should just call it that to avoid others from getting confused too.

But we are not adding support for pipes :)